dotnet/src/Microsoft.Agents.AI.AGUI/Shared/AIToolExtensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.AI;
+
+#if ASPNETCORE
+namespace Microsoft.Agents.AI.Hosting.AGUI.AspNetCore.Shared;
+#else
+namespace Microsoft.Agents.AI.AGUI.Shared;
+#endif
+
+internal static class AIToolExtensions
+{
+    public static IEnumerable<AGUITool> AsAGUITools(this IEnumerable<AITool> tools)
+    {
+        if (tools is null)
+        {
+            yield break;
+        }
+
+        foreach (var tool in tools)
+        {
+            // Convert both AIFunctionDeclaration and AIFunction (which extends it) to AGUITool
```

Comment:
why do we not need to handle `AIFunction` here? Not all `AIFunctions` are `AIFunctionDeclaration`, so we need to handle it as well?

Reply:
This is for the client tools. I think it's a requirement that any client tool you register is an AIFunctionDeclaration if you want it to be invokable through AG-UI, as otherwise I don't see a way in which you can provide the Json Schema for the parameters.